### PR TITLE
Update RateLimiter to use dictionary of rate limits (fixes #437)

### LINKED ISSUE #437: Update rate limit system to use arbitrary time periods
Default development rate limits are now:

     20 requests every 1 seconds
    100 requests every 2 minutes 

Should change the rate limiting system to support an arbitrary set of rate limits/time periods.

### PR DESCRIPTION
**Pull request by MingweiSamuel**
image

Tests pass locally (don't have tournament key). Not sure if CI is updated for the new system so...

### CODE REVIEW

File: RiotSharp.Test/appsettings.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿{
-  "ApiKey": "API_KEY",
+  "ApiKey": "RGAPI-369f27ac-9250-4e30-95e1-6fa1477f0048",
```
Comment (BenFradet) [inline RiotSharp.Test/appsettings.json]: !!

Comment (MingweiSamuel) [inline RiotSharp.Test/appsettings.json]: Lul

File: RiotSharp.AspNetCore/RiotSharpOptions.cs [hunk 2]
```diff
@@ -9,16 +9,12 @@ public class RiotSharpOptions
         public RiotSharpOptions()
         {
             RiotApi = new ApiKeyOptions();
-            RiotApi.RateLimitPer10S = 10;
-            RiotApi.RateLimitPer10M = 500;
             TournamentApi = new TournamentApiKeyOptions();
```
Comment (JanOuborny) [inline RiotSharp.AspNetCore/RiotSharpOptions.cs]: Can you add default values here?

**Comment by MingweiSamuel** [comment]
Removed the api key, added default RiotApi rate limits for the options. I don't have any experience with the Tournament API so I don't how those rate limits are. I figure that API isn't meant to be spammed anyway.

**Comment by JanOuborny** [comment]
@MingweiSamuel  The default rate limits for my tournament key are 10 per 10 seconds and 500 per 10 minutes. Can you also add these?

**Comment by MingweiSamuel** [comment]
Updated

**Comment by BenFradet** [comment]
Merging, thanks!

**Pull request merged**